glslc/src/main.cc

```diff
@@ -184,6 +192,49 @@ int main(int argc, char** argv) {
     } else if (arg == "-E") {
       compiler.options().SetPreprocessingOnlyMode();
       compiler.SetPreprocessingOnlyFlag();
+    } else if (arg == "-M" || arg == "-MM") {
+      // -M implies -E and -w
+      compiler.options().SetPreprocessingOnlyMode();
```

Comment:
It seems like an API problem that you'd have to call both options().SetPreprocessingOnlyMode *and* SetPreprocessingOnlyFlag.
It seems that you should only have to call SetPreprocessingOnlyFlag and it should take care of updating its internal options object.

But not for this CL.  It's a pre-existing problem.